src/envoy/http/service_control/client_cache_test.cc

```diff
@@ -448,6 +458,41 @@ TEST_F(ClientCacheCheckHttpRequestTest, OnePendingHttpCallCancelled) {
   checkAndReset(stats_base_.stats().filter_.denied_producer_error_, 1);
 }
 
+// Check call 1: Cache miss occurs, so cache makes HttpCall to SC Check.
+// HttpCall is successful, and the onCheckDone callback is called.
+// Check call 2 & 3: Cache hit, the CheckDoneFunc is called again.
+TEST_F(ClientCacheCheckHttpRequestTest, SuccessfulHttpCallWithCache) {
+  // Initially, we should only expect 1 HTTP call. But when a cached items is
+  // flushed, another http call is made. For this test, the flush happens during
+  // destruction.
+  setupHttpMocks(2);
+
+  CheckDoneFunc on_check_done = [this](const Status& got_status,
+                                       const CheckResponseInfo&) {
+    got_num_callbacks_++;
+    EXPECT_EQ(got_status.code(), Code::OK);
+  };
+
+  // Check call 1.
+  const CheckRequest request = getValidCheckRequest();
+  cache_->callCheck(request, mock_parent_span_, on_check_done);
+
+  // Stimulate successful http response.
+  // Test tear down will check the check callback is invoked.
+  std::string response_body;
+  const CheckResponse response = getValidCheckResponse();
+  response.SerializeToString(&response_body);
+  http_done_(Status::OK, response_body);
+
+  // Check call 2 & 3.
+  cache_->callCheck(request, mock_parent_span_, on_check_done);
```

Comment:
2, and 3 should be cache hit.  remote_http should not be called.

Reply:
That's exactly what this test shows. `setupHttpMocks` dictates the number of remote http calls.

`http_done_(...)` is  called to finish the http call for the first `callCheck`.

Then the 2nd and 3rd `callCheck` do not make any more remote http calls.